src/plugins/shared-variables/graph/plotted-variables-adornment/plotted-variables-adornment-component.tsx

```diff
@@ -215,50 +208,83 @@ export const PlottedVariablesAdornmentComponent = observer(function PlottedVaria
     return mstAutorun(() => {
       model.updateCategories(graphModel.layers[0].getUpdateCategoriesOptions(false));
     }, { name: "PlottedVariablesAdornmentComponent.refreshExpressionChange" }, model);
-  }, [graphModel, model, xScale, xSubAxesCount, yScale]);
+  }, [graphModel, model]);
 
-  // Refresh values on axis or expression change
+  // Draw when becoming visible
+  useEffect(function refreshOnVisibility() {
+    return mstReaction(
+      () => { return model.isVisible; },
+      (visible) => {
+        if (visible) {
+          refreshValues();
+        }
+      },
+      { name: "PlottedVariablesAdornment.refreshOnVisibility" },
+      model);
+  }, [model, refreshValues]);
+
+  // Refresh display when axis domains change
   useEffect(function refreshAxisChange() {
-    return mstAutorun(() => {
-      // We observe changes to the axis domains within the autorun by extracting them from the axes below.
-      // We do this instead of including domains in the useEffect dependency array to prevent domain changes
-      // from triggering a reinstall of the autorun.
-      if (xAxis && yAxis) {
-        const { domain: xDomain } = xAxis; // eslint-disable-line unused-imports/no-unused-vars
-        const { domain: yDomain } = yAxis; // eslint-disable-line unused-imports/no-unused-vars
-      }
-      // Trigger an autorun if any inputs or the expression of y change, or if the x variable changes
-      Array.from(model.plottedVariables.values()).forEach(plottedVariables => {
-        plottedVariables.yVariable?.computedValueIncludingMessageAndError; // eslint-disable-line no-unused-expressions
-        plottedVariables.xVariable; // eslint-disable-line no-unused-expressions
-      });
-      refreshValues();
-    }, { name: "PlottedVariablesAdornmentComponent.refreshAxisChange" }, model);
-  }, [dataConfig, model, plotWidth, plotHeight, sharedVariables, xAxis, yAxis, refreshValues]);
+    return mstReaction<string,boolean>(
+      () => {
+        if (xAxis && yAxis) {
+          const { domain: xDomain } = xAxis;
+          const { domain: yDomain } = yAxis;
+          // Return a primitive value (string, not array) so that equality will work
+          // and effect will only be called when these numbers actually change.
+          return `${xDomain[0]}, ${xDomain[1]}, ${yDomain[0]}, ${yDomain[1]}`;
```

Comment:
Could also just return an array or object and pass `equals: comparer.structural` in options.